storage/remote/queue_manager.go

```diff
@@ -777,6 +779,7 @@ func (s *shards) runShard(ctx context.Context, shardID int, queue chan sample) {
 	for {
 		select {
 		case <-ctx.Done():
+			s.qm.metrics.pendingSamples.Sub(float64(nPending + len(queue)))
```

Comment:
My only concern is that this + `Set(0)` in start would just hide the actual issue. Maybe we should have another metric that's "samples dropped because their shard was shutdown but was unable to flush it's samples within the timeout"?

Reply:
Good call, I have now added another line after this to add the dropped samples to the failed samples metric. Does that seem reasonable to you?

Reply:
I think that should be okay, since we also increment dropped samples for samples that we don't see a labels cache entry for (no entry for ref ID). But we should possibly also log this?

Reply:
We already have the log for `Failed to flush all samples on shutdown`, perhaps move that log line here? Or are you thinking a separate log line?

Reply:
Hmm. I think that log line is enough but it would be nice to add the # of samples we failed to flush to that log line. If that's a non-trivial change then we can just merge this as is and file that as a nice to have in the remote write meta issue.

Reply:
:+1: I added a commit with one way of adding the # of samples to the log line. I thought it would be the least invasive, but happy to try something else too.

Reply:
@cstyan bump for another review on the most recent changes when you get a chance!